### PR TITLE
fixed project for bucket datasource

### PR DESCRIPTION
**Pull request by edwardmedia**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/13034

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
storage: fixed the wrong `project` value on the data source of `google_storage_bucket` 
```

### CODE REVIEW

File: mmv1/third_party/terraform/data_sources/data_source_google_storage_bucket.go [hunk 1]
```diff
@@ -33,5 +35,8 @@ func dataSourceGoogleStorageBucketRead(d *schema.ResourceData, meta interface{})
 	}
 	log.Printf("[DEBUG] Read bucket %v at location %v\n\n", res.Name, res.SelfLink)
 
-	return setStorageBucket(d, config, res, bucket, userAgent)
+	setStorageBucket(d, config, res, bucket, userAgent)
+
+	d.Set("project", strconv.Itoa(int(res.ProjectNumber)))
```
Comment (martinOrigin8Cares) [inline mmv1/third_party/terraform/data_sources/data_source_google_storage_bucket.go:40]: Be careful here, you return the projectNumber and the documentation said that project is the projectId. Also in all data.source adn resource project represent always a project_id. You shuld convert hte projectNumber into a projectId

**Review by martinOrigin8Cares — CHANGES_REQUESTED** [review]
shoudl convert the projectNumber into a projectId

File: mmv1/third_party/terraform/data_sources/data_source_google_storage_bucket.go (hunk 1 above)
Comment (edwardmedia) [reply mmv1/third_party/terraform/data_sources/data_source_google_storage_bucket.go:40]: Was thought of that. But bucket is global, and the project_id is not used anywhere. Here is the api endpoint

```
GET /storage/v1/b/hostname-bucket?alt=json&prettyPrint=false HTTP/1.1
Host: storage.googleapis.com
```

And api returns [projectNumber](https://cloud.google.com/storage/docs/json_api/v1/buckets)

I think in the resource & datasource itself, it should have used project_number to match the API's.

Comment (martinOrigin8Cares) [reply mmv1/third_party/terraform/data_sources/data_source_google_storage_bucket.go:40]: Agree but when you create a bucket you need to specify a project (for only the billing purpose) even if it is global. Also if you don't specify the `project` in the provider configuration, the `data.google_storage_bucket` returns the correct project_id. So somebody already did the job but it is not consistent. 

In the case of the provider_google, it seems that they concentrate on using only project_id (almost everywhere, for at least the attribute `project` and used `project_number` in other case) 

So in this case, I will argue that project should be a `project_id` regardless of the google API

Comment (slevenick) [reply mmv1/third_party/terraform/data_sources/data_source_google_storage_bucket.go:40]: The problem is we can't convert project # -> project ID within the provider.

We have a default project ID from the provider-level configuration, but we can't tell if that matches a given project number

Comment (slevenick) [reply mmv1/third_party/terraform/data_sources/data_source_google_storage_bucket.go:40]: That said, I don't think we should set the project to a number at all. It is probably preferable to just not set this

Comment (slevenick) [reply mmv1/third_party/terraform/data_sources/data_source_google_storage_bucket.go:40]: Hm, I guess we can & do this in setStorageBucket. I just don't really think anything is wrong here as the bucket name is a unique identifier, so when we have that the project isn't used during lookup.

Comment (edwardmedia) [reply mmv1/third_party/terraform/data_sources/data_source_google_storage_bucket.go:40]: Yeah, looking at below lib, there isn't a query available that can be used to convert project # -> project ID 

https://pkg.go.dev/google.golang.org/api/cloudresourcemanager/v3

**Comment by edwardmedia** [comment]
Closing this

**Pull request closed without merging**